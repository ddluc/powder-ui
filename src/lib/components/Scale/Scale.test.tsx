import React from 'react';
import { cleanup } from '@testing-library/react';
import { Scale, ScaleProps } from './index';
import withTheme from '../../../../test/withTheme';

// Define unit tests
describe('Scale', () => {
  afterEach(cleanup);

  const args: ScaleProps = {
    lg: '1200px',
    md: '800px',
    sm: '200px',
    xs: '80%'
  };

  it('should render with defined props', () => {
    const result = withTheme(<Scale {...args} />);
    const component = result.container.firstChild;
    expect(component).toMatchSnapshot();
  });
});
