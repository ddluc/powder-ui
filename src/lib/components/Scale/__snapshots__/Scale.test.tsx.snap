// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scale should render with defined props 1`] = `
.c0 {
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

.c1 {
  width: 1200px;
}

@media screen and (max-width:960px) {
  .c1 {
    width: 800px;
  }
}

@media screen and (max-width:720px) {
  .c1 {
    width: 200px;
  }
}

@media screen and (max-width:400px) {
  .c1 {
    width: 80%;
  }
}

<div
  class="c0 c1"
/>
`;
